x-pack/osquerybeat/cmd/root.go

```diff
@@ -74,14 +75,30 @@ func genVerifyCmd(_ instance.Settings) *cobra.Command {
 func osquerybeatCfg(rawIn *proto.UnitExpectedConfig, agentInfo *client.AgentInfo) ([]*reload.ConfigWithMeta, error) {
 	// Convert to streams, osquerybeat doesn't use streams
 	streams := make([]*proto.Stream, 1)
+
+	// Enforce the datastream dataset and type because the libbeat call to CreateInputsFromStreams
+	// provides it's own defaults that are breaking the osquery with logstash
+	// The target datastream for the publisher is expected to be logs-osquery_manager.result-<namespace>
+	// while the libebeat management.CreateInputsFromStreams defaults to osquery-generic-default
+	var datastream *proto.DataStream
+	if rawIn.GetDataStream() != nil {
+		// Copy by value and modify dataset and type
+		ds := *rawIn.GetDataStream()
+		ds.Dataset = config.DefaultDataset
+		ds.Type = config.DefaultType
+		datastream = &ds
+	}
+
 	streams[0] = &proto.Stream{
 		Source:     rawIn.GetSource(),
 		Id:         rawIn.GetId(),
-		DataStream: rawIn.GetDataStream(),
+		DataStream: datastream,
 	}
+
 	rawIn.Streams = streams
 
 	procs := defaultProcessors()
+
```

Comment:
I can't comment directly on this, but this block below can be removed:

```go
	for iter := range modules {
		modules[iter]["type"] = "log"
	}
```

That is trying to set an input type, which I don't think matters for osquerybeat. It is just adding an arbitrary `type: log` into the generated config that osquerybeat will be ignoring.

Reply:
yeah, don't remember the reason for this. can clean up later, separately from this more urgent fix

Reply:
I think it is a copy/paste error from one of the other Beats, agree this can be removed separately.